bootstrap.sh

```diff
@@ -114,7 +122,7 @@ function build_osrm() {
     if [[ ${NM:-false} != false ]]; then
         CMAKE_EXTRA_ARGS="${CMAKE_EXTRA_ARGS} -DCMAKE_NM=${NM}"
     fi
-    ../../../mason_packages/.link/bin/cmake ../ -DCMAKE_INSTALL_PREFIX=${MASON_HOME} \
+    ${MASON_HOME}/bin/cmake ../ -DCMAKE_INSTALL_PREFIX=${MASON_HOME} \
```

Comment:
I just took a look at the CMake options, and while we're setting `-DCMAKE_EXE_LINKER_FLAGS="${LINK_FLAGS}"` do we need to set `CMAKE_SHARED_LINKER_FLAGS` for `libosrm.so` in addition?